# Publish all joint states -----------------------------------
joint_state_controller:
  type: joint_state_controller/JointStateController
  publish_rate: 50

# Position Controllers ---------------------------------------
lleg_crotch_p_position:
  type: effort_controllers/JointPositionController
  joint: lleg_crotch_p
  pid: {p: 400000000.000000, i: 4000000.000000, d: 500000.000000}
  mimic_joints:
    lleg_crotch_p_front_mimic_position:
      type: effort_controllers/JointPositionController
      joint: lleg_crotch_p_front_mimic
      pid: {p: 50000.000000, i: 500.000000, d: 5000.000000}
    lleg_crotch_p_back_mimic_position:
      type: effort_controllers/JointPositionController
      joint: lleg_crotch_p_back_mimic
      pid: {p: 50000.000000, i: 500.000000, d: 5000.000000}
torso_rthrust_p_position:
  type: effort_controllers/JointPositionController
  joint: torso_rthrust_p
  pid: {p: 10000000.000000, i: 500.000000, d: 20000.000000}
lleg_knee_p2_position:
  type: effort_controllers/JointPositionController
  joint: lleg_knee_p2
  pid: {p: 200000000.000000, i: 1000000.000000, d: 100000.000000}
lleg_ankle_p_position:
  type: effort_controllers/JointPositionController
  joint: lleg_ankle_p
  pid: {p: 100000000.000000, i: 5000.000000, d: 500000.000000}
  mimic_joints:
    lleg_ankle_p_mimic_position:
      type: effort_controllers/JointPositionController
      joint: lleg_ankle_p_mimic
      pid: {p: 500000.000000, i: 500.000000, d: 50000.000000}
lleg_crotch_r_position:
  type: effort_controllers/JointPositionController
  joint: lleg_crotch_r
  pid: {p: 200000000.000000, i: 1000000.000000, d: 500000.000000}
  mimic_joints:
    lleg_crotch_r_mimic_position:
      type: effort_controllers/JointPositionController
      joint: lleg_crotch_r_mimic
      pid: {p: 50000.000000, i: 500.000000, d: 5000.000000}
lleg_crotch_y_position:
  type: effort_controllers/JointPositionController
  joint: lleg_crotch_y
  pid: {p: 200000000.000000, i: 1000000.000000, d: 500000.000000}
lleg_knee_p_position:
  type: effort_controllers/JointPositionController
  joint: lleg_knee_p
  pid: {p: 200000000.000000, i: 1000000.000000, d: 100000.000000}
rarm_shoulder_p_position:
  type: effort_controllers/JointPositionController
  joint: rarm_shoulder_p
  pid: {p: 10000000.000000, i: 500.000000, d: 200000.000000}
lleg_ankle_r_position:
  type: effort_controllers/JointPositionController
  joint: lleg_ankle_r
  pid: {p: 100000000.000000, i: 5000.000000, d: 500000.000000}
  mimic_joints:
    lleg_ankle_r_mimic_position:
      type: effort_controllers/JointPositionController
      joint: lleg_ankle_r_mimic
      pid: {p: 100000000.000000, i: 1000000.000000, d: 50000.000000}
torso_lthrust_p_position:
  type: effort_controllers/JointPositionController
  joint: torso_lthrust_p
  pid: {p: 10000000.000000, i: 500.000000, d: 20000.000000}
rarm_elbow_p_position:
  type: effort_controllers/JointPositionController
  joint: rarm_elbow_p
  pid: {p: 20000000.000000, i: 1000.000000, d: 40000.000000}
rarm_elbow_p2_position:
  type: effort_controllers/JointPositionController
  joint: rarm_elbow_p2
  pid: {p: 20000000.000000, i: 1000.000000, d: 40000.000000}
rarm_shoulder_r_position:
  type: effort_controllers/JointPositionController
  joint: rarm_shoulder_r
  pid: {p: 10000000.000000, i: 500.000000, d: 200000.000000}
rarm_shoulder_y_position:
  type: effort_controllers/JointPositionController
  joint: rarm_shoulder_y
  pid: {p: 10000000.000000, i: 500.000000, d: 200000.000000}
rarm_gripper_position:
  type: effort_controllers/JointPositionController
  joint: rarm_gripper
  pid: {p: 100000.000000, i: 100.000000, d: 100.000000}
  mimic_joints:
    rarm_gripper_thumb2_mimic_position:
      type: effort_controllers/JointPositionController
      joint: rarm_gripper_thumb2_mimic
      pid: {p: 100000.000000, i: 100.000000, d: 100.000000}
    rarm_gripper_thumb1_mimic_position:
      type: effort_controllers/JointPositionController
      joint: rarm_gripper_thumb1_mimic
      pid: {p: 100000.000000, i: 100.000000, d: 100.000000}
    rarm_gripper_ring2_mimic_position:
      type: effort_controllers/JointPositionController
      joint: rarm_gripper_ring2_mimic
      pid: {p: 100000.000000, i: 100.000000, d: 100.000000}
    rarm_gripper_ring0_mimic_position:
      type: effort_controllers/JointPositionController
      joint: rarm_gripper_ring0_mimic
      pid: {p: 100000.000000, i: 100.000000, d: 100.000000}
    rarm_gripper_ring1_mimic_position:
      type: effort_controllers/JointPositionController
      joint: rarm_gripper_ring1_mimic
      pid: {p: 100000.000000, i: 100.000000, d: 100.000000}
    rarm_gripper_middle2_mimic_position:
      type: effort_controllers/JointPositionController
      joint: rarm_gripper_middle2_mimic
      pid: {p: 100000.000000, i: 100.000000, d: 100.000000}
    rarm_gripper_middle1_mimic_position:
      type: effort_controllers/JointPositionController
      joint: rarm_gripper_middle1_mimic
      pid: {p: 100000.000000, i: 100.000000, d: 100.000000}
    rarm_gripper_middle0_mimic_position:
      type: effort_controllers/JointPositionController
      joint: rarm_gripper_middle0_mimic
      pid: {p: 100000.000000, i: 100.000000, d: 100.000000}
    rarm_gripper_little0_mimic_position:
      type: effort_controllers/JointPositionController
      joint: rarm_gripper_little0_mimic
      pid: {p: 100000.000000, i: 100.000000, d: 100.000000}
    rarm_gripper_index2_mimic_position:
      type: effort_controllers/JointPositionController
      joint: rarm_gripper_index2_mimic
      pid: {p: 100000.000000, i: 100.000000, d: 100.000000}
    rarm_gripper_index1_mimic_position:
      type: effort_controllers/JointPositionController
      joint: rarm_gripper_index1_mimic
      pid: {p: 100000.000000, i: 100.000000, d: 100.000000}
    rarm_gripper_index0_mimic_position:
      type: effort_controllers/JointPositionController
      joint: rarm_gripper_index0_mimic
      pid: {p: 100000.000000, i: 100.000000, d: 100.000000}
    rarm_gripper_little2_mimic_position:
      type: effort_controllers/JointPositionController
      joint: rarm_gripper_little2_mimic
      pid: {p: 100000.000000, i: 100.000000, d: 100.000000}
    rarm_gripper_little1_mimic_position:
      type: effort_controllers/JointPositionController
      joint: rarm_gripper_little1_mimic
      pid: {p: 100000.000000, i: 100.000000, d: 100.000000}
rarm_wrist_y_position:
  type: effort_controllers/JointPositionController
  joint: rarm_wrist_y
  pid: {p: 100000.000000, i: 50.000000, d: 20000.000000}
rarm_wrist_r_position:
  type: effort_controllers/JointPositionController
  joint: rarm_wrist_r
  pid: {p: 100000.000000, i: 50.000000, d: 20000.000000}
larm_wrist_y_position:
  type: effort_controllers/JointPositionController
  joint: larm_wrist_y
  pid: {p: 100000.000000, i: 50.000000, d: 20000.000000}
larm_elbow_p2_position:
  type: effort_controllers/JointPositionController
  joint: larm_elbow_p2
  pid: {p: 20000000.000000, i: 1000.000000, d: 40000.000000}
larm_gripper_position:
  type: effort_controllers/JointPositionController
  joint: larm_gripper
  pid: {p: 100000.000000, i: 100.000000, d: 100.000000}
  mimic_joints:
    larm_gripper_index0_mimic_position:
      type: effort_controllers/JointPositionController
      joint: larm_gripper_index0_mimic
      pid: {p: 100000.000000, i: 100.000000, d: 100.000000}
    larm_gripper_thumb2_mimic_position:
      type: effort_controllers/JointPositionController
      joint: larm_gripper_thumb2_mimic
      pid: {p: 100000.000000, i: 100.000000, d: 100.000000}
    larm_gripper_thumb1_mimic_position:
      type: effort_controllers/JointPositionController
      joint: larm_gripper_thumb1_mimic
      pid: {p: 100000.000000, i: 100.000000, d: 100.000000}
    larm_gripper_index2_mimic_position:
      type: effort_controllers/JointPositionController
      joint: larm_gripper_index2_mimic
      pid: {p: 100000.000000, i: 100.000000, d: 100.000000}
    larm_gripper_index1_mimic_position:
      type: effort_controllers/JointPositionController
      joint: larm_gripper_index1_mimic
      pid: {p: 100000.000000, i: 100.000000, d: 100.000000}
    larm_gripper_middle0_mimic_position:
      type: effort_controllers/JointPositionController
      joint: larm_gripper_middle0_mimic
      pid: {p: 100000.000000, i: 100.000000, d: 100.000000}
    larm_gripper_middle1_mimic_position:
      type: effort_controllers/JointPositionController
      joint: larm_gripper_middle1_mimic
      pid: {p: 100000.000000, i: 100.000000, d: 100.000000}
    larm_gripper_middle2_mimic_position:
      type: effort_controllers/JointPositionController
      joint: larm_gripper_middle2_mimic
      pid: {p: 100000.000000, i: 100.000000, d: 100.000000}
    larm_gripper_ring0_mimic_position:
      type: effort_controllers/JointPositionController
      joint: larm_gripper_ring0_mimic
      pid: {p: 100000.000000, i: 100.000000, d: 100.000000}
    larm_gripper_ring1_mimic_position:
      type: effort_controllers/JointPositionController
      joint: larm_gripper_ring1_mimic
      pid: {p: 100000.000000, i: 100.000000, d: 100.000000}
    larm_gripper_ring2_mimic_position:
      type: effort_controllers/JointPositionController
      joint: larm_gripper_ring2_mimic
      pid: {p: 100000.000000, i: 100.000000, d: 100.000000}
    larm_gripper_little0_mimic_position:
      type: effort_controllers/JointPositionController
      joint: larm_gripper_little0_mimic
      pid: {p: 100000.000000, i: 100.000000, d: 100.000000}
    larm_gripper_little1_mimic_position:
      type: effort_controllers/JointPositionController
      joint: larm_gripper_little1_mimic
      pid: {p: 100000.000000, i: 100.000000, d: 100.000000}
    larm_gripper_little2_mimic_position:
      type: effort_controllers/JointPositionController
      joint: larm_gripper_little2_mimic
      pid: {p: 100000.000000, i: 100.000000, d: 100.000000}
larm_wrist_r_position:
  type: effort_controllers/JointPositionController
  joint: larm_wrist_r
  pid: {p: 100000.000000, i: 50.000000, d: 20000.000000}
torso_rthrust_r_position:
  type: effort_controllers/JointPositionController
  joint: torso_rthrust_r
  pid: {p: 10000000.000000, i: 500.000000, d: 20000.000000}
larm_shoulder_p_position:
  type: effort_controllers/JointPositionController
  joint: larm_shoulder_p
  pid: {p: 10000000.000000, i: 500.000000, d: 200000.000000}
torso_lthrust_r_position:
  type: effort_controllers/JointPositionController
  joint: torso_lthrust_r
  pid: {p: 10000000.000000, i: 500.000000, d: 20000.000000}
larm_shoulder_r_position:
  type: effort_controllers/JointPositionController
  joint: larm_shoulder_r
  pid: {p: 10000000.000000, i: 500.000000, d: 200000.000000}
larm_elbow_p_position:
  type: effort_controllers/JointPositionController
  joint: larm_elbow_p
  pid: {p: 20000000.000000, i: 1000.000000, d: 40000.000000}
larm_shoulder_y_position:
  type: effort_controllers/JointPositionController
  joint: larm_shoulder_y
  pid: {p: 10000000.000000, i: 500.000000, d: 200000.000000}
rleg_ankle_p_position:
  type: effort_controllers/JointPositionController
  joint: rleg_ankle_p
  pid: {p: 100000000.000000, i: 5000.000000, d: 500000.000000}
  mimic_joints:
    rleg_ankle_p_mimic_position:
      type: effort_controllers/JointPositionController
      joint: rleg_ankle_p_mimic
      pid: {p: 500000.000000, i: 500.000000, d: 50000.000000}
rleg_knee_p2_position:
  type: effort_controllers/JointPositionController
  joint: rleg_knee_p2
  pid: {p: 200000000.000000, i: 1000000.000000, d: 100000.000000}
rleg_crotch_r_position:
  type: effort_controllers/JointPositionController
  joint: rleg_crotch_r
  pid: {p: 200000000.000000, i: 1000000.000000, d: 500000.000000}
  mimic_joints:
    rleg_crotch_r_mimic_position:
      type: effort_controllers/JointPositionController
      joint: rleg_crotch_r_mimic
      pid: {p: 50000.000000, i: 500.000000, d: 5000.000000}
rleg_knee_p_position:
  type: effort_controllers/JointPositionController
  joint: rleg_knee_p
  pid: {p: 200000000.000000, i: 1000000.000000, d: 100000.000000}
rleg_crotch_y_position:
  type: effort_controllers/JointPositionController
  joint: rleg_crotch_y
  pid: {p: 200000000.000000, i: 1000000.000000, d: 500000.000000}
torso_waist_p2_position:
  type: effort_controllers/JointPositionController
  joint: torso_waist_p2
  pid: {p: 20000000.000000, i: 10000.000000, d: 1000000.000000}
torso_waist_p_position:
  type: effort_controllers/JointPositionController
  joint: torso_waist_p
  pid: {p: 20000000.000000, i: 10000.000000, d: 1000000.000000}
head_neck_p_position:
  type: effort_controllers/JointPositionController
  joint: head_neck_p
  pid: {p: 1000000.000000, i: 500.000000, d: 200000.000000}
head_neck_y_position:
  type: effort_controllers/JointPositionController
  joint: head_neck_y
  pid: {p: 1000000.000000, i: 500.000000, d: 200000.000000}
torso_waist_y_position:
  type: effort_controllers/JointPositionController
  joint: torso_waist_y
  pid: {p: 20000000.000000, i: 10000.000000, d: 1000000.000000}
rleg_ankle_r_position:
  type: effort_controllers/JointPositionController
  joint: rleg_ankle_r
  pid: {p: 100000000.000000, i: 5000.000000, d: 500000.000000}
  mimic_joints:
    rleg_ankle_r_mimic_position:
      type: effort_controllers/JointPositionController
      joint: rleg_ankle_r_mimic
      pid: {p: 100000000.000000, i: 1000000.000000, d: 50000.000000}
rleg_crotch_p_position:
  type: effort_controllers/JointPositionController
  joint: rleg_crotch_p
  pid: {p: 400000000.000000, i: 4000000.000000, d: 500000.000000}
  mimic_joints:
    rleg_crotch_p_back_mimic_position:
      type: effort_controllers/JointPositionController
      joint: rleg_crotch_p_back_mimic
      pid: {p: 50000.000000, i: 500.000000, d: 5000.000000}
    rleg_crotch_p_front_mimic_position:
      type: effort_controllers/JointPositionController
      joint: rleg_crotch_p_front_mimic
      pid: {p: 50000.000000, i: 500.000000, d: 5000.000000}

# Joint Trajectory Controllers ---------------------------------------
fullbody_controller:
#  type: effort_controllers/JointTrajectoryController
  type: gundam_rx78_control/JointTrajectoryController
  joints:
    - lleg_crotch_p
    - torso_rthrust_p
    - lleg_knee_p2
    - lleg_ankle_p
    - lleg_crotch_r
    - lleg_crotch_y
    - lleg_knee_p
    - rarm_shoulder_p
    - lleg_ankle_r
    - torso_lthrust_p
    - rarm_elbow_p
    - rarm_elbow_p2
    - rarm_shoulder_r
    - rarm_shoulder_y
    - rarm_gripper
    - rarm_wrist_y
    - rarm_wrist_r
    - larm_wrist_y
    - larm_elbow_p2
    - larm_gripper
    - larm_wrist_r
    - torso_rthrust_r
    - larm_shoulder_p
    - torso_lthrust_r
    - larm_shoulder_r
    - larm_elbow_p
    - larm_shoulder_y
    - rleg_ankle_p
    - rleg_knee_p2
    - rleg_crotch_r
    - rleg_knee_p
    - rleg_crotch_y
    - torso_waist_p2
    - torso_waist_p
    - head_neck_p
    - head_neck_y
    - torso_waist_y
    - rleg_ankle_r
    - rleg_crotch_p
  mimic_joints:
    - lleg_crotch_p_front_mimic # lleg_crotch_p
    - rleg_ankle_p_mimic # rleg_ankle_p
    - rleg_crotch_p_back_mimic # rleg_crotch_p
    - rleg_crotch_r_mimic # rleg_crotch_r
    - lleg_crotch_p_back_mimic # lleg_crotch_p
    - lleg_ankle_r_mimic # lleg_ankle_r
    - lleg_crotch_r_mimic # lleg_crotch_r
    - lleg_ankle_p_mimic # lleg_ankle_p
    - rarm_gripper_thumb2_mimic # rarm_gripper
    - larm_gripper_index0_mimic # larm_gripper
    - rarm_gripper_thumb1_mimic # rarm_gripper
    - larm_gripper_thumb2_mimic # larm_gripper
    - larm_gripper_thumb1_mimic # larm_gripper
    - larm_gripper_index2_mimic # larm_gripper
    - larm_gripper_index1_mimic # larm_gripper
    - larm_gripper_middle0_mimic # larm_gripper
    - rleg_crotch_p_front_mimic # rleg_crotch_p
    - rarm_gripper_ring2_mimic # rarm_gripper
    - larm_gripper_middle1_mimic # larm_gripper
    - larm_gripper_middle2_mimic # larm_gripper
    - larm_gripper_ring0_mimic # larm_gripper
    - larm_gripper_ring1_mimic # larm_gripper
    - larm_gripper_ring2_mimic # larm_gripper
    - larm_gripper_little0_mimic # larm_gripper
    - larm_gripper_little1_mimic # larm_gripper
    - larm_gripper_little2_mimic # larm_gripper
    - rarm_gripper_ring0_mimic # rarm_gripper
    - rarm_gripper_ring1_mimic # rarm_gripper
    - rleg_ankle_r_mimic # rleg_ankle_r
    - rarm_gripper_middle2_mimic # rarm_gripper
    - rarm_gripper_middle1_mimic # rarm_gripper
    - rarm_gripper_middle0_mimic # rarm_gripper
    - rarm_gripper_little0_mimic # rarm_gripper
    - rarm_gripper_index2_mimic # rarm_gripper
    - rarm_gripper_index1_mimic # rarm_gripper
    - rarm_gripper_index0_mimic # rarm_gripper
    - rarm_gripper_little2_mimic # rarm_gripper
    - rarm_gripper_little1_mimic # rarm_gripper
  gains:
    lleg_crotch_p : {p: 400000000.000000, i: 4000000.000000, d: 500000.000000}
    torso_rthrust_p : {p: 10000000.000000, i: 500.000000, d: 20000.000000}
    lleg_knee_p2 : {p: 200000000.000000, i: 1000000.000000, d: 100000.000000}
    lleg_ankle_p : {p: 100000000.000000, i: 5000.000000, d: 500000.000000}
    lleg_crotch_r : {p: 200000000.000000, i: 1000000.000000, d: 500000.000000}
    lleg_crotch_y : {p: 200000000.000000, i: 1000000.000000, d: 500000.000000}
    lleg_knee_p : {p: 200000000.000000, i: 1000000.000000, d: 100000.000000}
    rarm_shoulder_p : {p: 10000000.000000, i: 500.000000, d: 200000.000000}
    lleg_ankle_r : {p: 100000000.000000, i: 5000.000000, d: 500000.000000}
    torso_lthrust_p : {p: 10000000.000000, i: 500.000000, d: 20000.000000}
    rarm_elbow_p : {p: 20000000.000000, i: 1000.000000, d: 40000.000000}
    rarm_elbow_p2 : {p: 20000000.000000, i: 1000.000000, d: 40000.000000}
    rarm_shoulder_r : {p: 10000000.000000, i: 500.000000, d: 200000.000000}
    rarm_shoulder_y : {p: 10000000.000000, i: 500.000000, d: 200000.000000}
    rarm_gripper : {p: 100000.000000, i: 100.000000, d: 100.000000}
    rarm_wrist_y : {p: 100000.000000, i: 50.000000, d: 20000.000000}
    rarm_wrist_r : {p: 100000.000000, i: 50.000000, d: 20000.000000}
    larm_wrist_y : {p: 100000.000000, i: 50.000000, d: 20000.000000}
    larm_elbow_p2 : {p: 20000000.000000, i: 1000.000000, d: 40000.000000}
    larm_gripper : {p: 100000.000000, i: 100.000000, d: 100.000000}
    larm_wrist_r : {p: 100000.000000, i: 50.000000, d: 20000.000000}
    torso_rthrust_r : {p: 10000000.000000, i: 500.000000, d: 20000.000000}
    larm_shoulder_p : {p: 10000000.000000, i: 500.000000, d: 200000.000000}
    torso_lthrust_r : {p: 10000000.000000, i: 500.000000, d: 20000.000000}
    larm_shoulder_r : {p: 10000000.000000, i: 500.000000, d: 200000.000000}
    larm_elbow_p : {p: 20000000.000000, i: 1000.000000, d: 40000.000000}
    larm_shoulder_y : {p: 10000000.000000, i: 500.000000, d: 200000.000000}
    rleg_ankle_p : {p: 100000000.000000, i: 5000.000000, d: 500000.000000}
    rleg_knee_p2 : {p: 200000000.000000, i: 1000000.000000, d: 100000.000000}
    rleg_crotch_r : {p: 200000000.000000, i: 1000000.000000, d: 500000.000000}
    rleg_knee_p : {p: 200000000.000000, i: 1000000.000000, d: 100000.000000}
    rleg_crotch_y : {p: 200000000.000000, i: 1000000.000000, d: 500000.000000}
    torso_waist_p2 : {p: 20000000.000000, i: 10000.000000, d: 1000000.000000}
    torso_waist_p : {p: 20000000.000000, i: 10000.000000, d: 1000000.000000}
    head_neck_p : {p: 1000000.000000, i: 500.000000, d: 200000.000000}
    head_neck_y : {p: 1000000.000000, i: 500.000000, d: 200000.000000}
    torso_waist_y : {p: 20000000.000000, i: 10000.000000, d: 1000000.000000}
    rleg_ankle_r : {p: 100000000.000000, i: 5000.000000, d: 500000.000000}
    rleg_crotch_p : {p: 400000000.000000, i: 4000000.000000, d: 500000.000000}
    lleg_crotch_p_front_mimic : {p: 50000.000000, i: 500.000000, d: 5000.000000}    # mimic joint of lleg_crotch_p_joint
    rleg_ankle_p_mimic : {p: 500000.000000, i: 500.000000, d: 50000.000000}    # mimic joint of rleg_ankle_p_joint
    rleg_crotch_p_back_mimic : {p: 50000.000000, i: 500.000000, d: 5000.000000}    # mimic joint of rleg_crotch_p_joint
    rleg_crotch_r_mimic : {p: 50000.000000, i: 500.000000, d: 5000.000000}    # mimic joint of rleg_crotch_r_joint
    lleg_crotch_p_back_mimic : {p: 50000.000000, i: 500.000000, d: 5000.000000}    # mimic joint of lleg_crotch_p_joint
    lleg_ankle_r_mimic : {p: 100000000.000000, i: 1000000.000000, d: 50000.000000}    # mimic joint of lleg_ankle_r_joint
    lleg_crotch_r_mimic : {p: 50000.000000, i: 500.000000, d: 5000.000000}    # mimic joint of lleg_crotch_r_joint
    lleg_ankle_p_mimic : {p: 500000.000000, i: 500.000000, d: 50000.000000}    # mimic joint of lleg_ankle_p_joint
    rarm_gripper_thumb2_mimic : {p: 100000.000000, i: 100.000000, d: 100.000000}    # mimic joint of rarm_gripper_joint
    larm_gripper_index0_mimic : {p: 100000.000000, i: 100.000000, d: 100.000000}    # mimic joint of larm_gripper_joint
    rarm_gripper_thumb1_mimic : {p: 100000.000000, i: 100.000000, d: 100.000000}    # mimic joint of rarm_gripper_joint
    larm_gripper_thumb2_mimic : {p: 100000.000000, i: 100.000000, d: 100.000000}    # mimic joint of larm_gripper_joint
    larm_gripper_thumb1_mimic : {p: 100000.000000, i: 100.000000, d: 100.000000}    # mimic joint of larm_gripper_joint
    larm_gripper_index2_mimic : {p: 100000.000000, i: 100.000000, d: 100.000000}    # mimic joint of larm_gripper_joint
    larm_gripper_index1_mimic : {p: 100000.000000, i: 100.000000, d: 100.000000}    # mimic joint of larm_gripper_joint
    larm_gripper_middle0_mimic : {p: 100000.000000, i: 100.000000, d: 100.000000}    # mimic joint of larm_gripper_joint
    rleg_crotch_p_front_mimic : {p: 50000.000000, i: 500.000000, d: 5000.000000}    # mimic joint of rleg_crotch_p_joint
    rarm_gripper_ring2_mimic : {p: 100000.000000, i: 100.000000, d: 100.000000}    # mimic joint of rarm_gripper_joint
    larm_gripper_middle1_mimic : {p: 100000.000000, i: 100.000000, d: 100.000000}    # mimic joint of larm_gripper_joint
    larm_gripper_middle2_mimic : {p: 100000.000000, i: 100.000000, d: 100.000000}    # mimic joint of larm_gripper_joint
    larm_gripper_ring0_mimic : {p: 100000.000000, i: 100.000000, d: 100.000000}    # mimic joint of larm_gripper_joint
    larm_gripper_ring1_mimic : {p: 100000.000000, i: 100.000000, d: 100.000000}    # mimic joint of larm_gripper_joint
    larm_gripper_ring2_mimic : {p: 100000.000000, i: 100.000000, d: 100.000000}    # mimic joint of larm_gripper_joint
    larm_gripper_little0_mimic : {p: 100000.000000, i: 100.000000, d: 100.000000}    # mimic joint of larm_gripper_joint
    larm_gripper_little1_mimic : {p: 100000.000000, i: 100.000000, d: 100.000000}    # mimic joint of larm_gripper_joint
    larm_gripper_little2_mimic : {p: 100000.000000, i: 100.000000, d: 100.000000}    # mimic joint of larm_gripper_joint
    rarm_gripper_ring0_mimic : {p: 100000.000000, i: 100.000000, d: 100.000000}    # mimic joint of rarm_gripper_joint
    rarm_gripper_ring1_mimic : {p: 100000.000000, i: 100.000000, d: 100.000000}    # mimic joint of rarm_gripper_joint
    rleg_ankle_r_mimic : {p: 100000000.000000, i: 1000000.000000, d: 50000.000000}    # mimic joint of rleg_ankle_r_joint
    rarm_gripper_middle2_mimic : {p: 100000.000000, i: 100.000000, d: 100.000000}    # mimic joint of rarm_gripper_joint
    rarm_gripper_middle1_mimic : {p: 100000.000000, i: 100.000000, d: 100.000000}    # mimic joint of rarm_gripper_joint
    rarm_gripper_middle0_mimic : {p: 100000.000000, i: 100.000000, d: 100.000000}    # mimic joint of rarm_gripper_joint
    rarm_gripper_little0_mimic : {p: 100000.000000, i: 100.000000, d: 100.000000}    # mimic joint of rarm_gripper_joint
    rarm_gripper_index2_mimic : {p: 100000.000000, i: 100.000000, d: 100.000000}    # mimic joint of rarm_gripper_joint
    rarm_gripper_index1_mimic : {p: 100000.000000, i: 100.000000, d: 100.000000}    # mimic joint of rarm_gripper_joint
    rarm_gripper_index0_mimic : {p: 100000.000000, i: 100.000000, d: 100.000000}    # mimic joint of rarm_gripper_joint
    rarm_gripper_little2_mimic : {p: 100000.000000, i: 100.000000, d: 100.000000}    # mimic joint of rarm_gripper_joint
    rarm_gripper_little1_mimic : {p: 100000.000000, i: 100.000000, d: 100.000000}    # mimic joint of rarm_gripper_joint
  constraints:
    goal_time: 0.6
    stopped_velocity_tolerance: 0.05
    # joint: {trajectory: 0.2, goal: 0.2}
  stop_trajectory_duration: 0.5
  state_publish_rate:  125
  action_monitor_rate: 10
  allow_partial_joints_goal: true
